api/envoy/extensions/filters/http/ext_proc/v3/processing_mode.proto

```diff
@@ -68,6 +69,34 @@ message ProcessingMode {
     // chunk. If the body exceeds the configured buffer limit, then the body contents
     // up to the buffer limit will be sent.
     BUFFERED_PARTIAL = 3;
+
+    // [#not-implemented-hide:]
+    // Envoy streams the body to the server in pieces as they arrive.
```

Comment:
Here's a suggested wording. I'm not sure if I fully captured the intent you had, so feel free to clarify it further (another option is to pass this through an NLP model).
```
    // Envoy streams the body to the server in pieces as they arrive.
    //
    // This mode allows Envoy and the server to send any number of body chunks to each other.
    // Envoy will send the body chunks as soon as they arrive, and the server may send any number of body chunks back to Envoy.
    // The last 
    // 
    // The individual response chunk size is recommended to be no greater than 64K bytes, or
    // :ref:`max_receive_message_length <envoy_v3_api_field_config.core.v3.GrpcService.EnvoyGrpc.max_receive_message_length>`
    // if EnvoyGrpc is used.
    //
    // Note that the server may also choose to buffer the entire message, including the headers (if header mode is
    // ``SEND``), the entire body, and the trailers (if present), before sending back any response.
    // The server response has to maintain the headers-body-trailers ordering.
    //
    // In this body mode:
    // * The corresponding trailer mode has to be set to ``SEND``.
    // * Envoy will send the body and trailers (if present) to the server as they arrive.
    //    Sending the trailers (if present) is to inform the server the complete body arrives.
    //    In case there are no no trailers, then Envoy will set
    //    :ref:`end_of_stream <envoy_v3_api_field_service.ext_proc.v3.HttpBody.end_of_stream>`
    //    as part of the last body chunk to notify the server that no other data is to be sent.
    // * The server needs to send
    //   :ref:`StreamedBodyResponse <envoy_v3_api_msg_service.ext_proc.v3.StreamedBodyResponse>`
    //   to Envoy in the body response.
    // * Envoy will stream the body chunks from the server to the upstream/downstream, as they arrive.
```

Reply:
I incorporated your 2nd part to the API. However, I kept my original part which describes the server buffering behavior, which is the essence of this mode.

Reply:
The reason I suggested an alternative wording is because there can be use-cases that are do not require buffering. Consider a processor that only cares about whether some substring appears in the given data, and for that buffering may not be needed.

Reply:
That sounds like a typical use case for the existing STREAMED mode.

Reply:
It's true that the server may not buffer even in the new MxN mode, like if M = 1, and N = 1, which is supported.  In such case, it's virtually working the same way as the existing STREAMED mode.   

However, that's not the typical use cases of MxN.

Reply:
MxN mode does support server process individual chunk, and sends back response. However,  I guess that's not the main case. That case can also be supported by STREAMED mode.

Also, maybe can have a follow up PR to discuss the API when we turn this on when removing // [#not-implemented-hide:]?

Reply:
I added a line to describe the case that server may not buffer and sends back a response after each request as 
//  3) Note that the server may also choose not to buffer. i.e, once it receives a body request,
//  it processes the data, and sends back a body response right away.

Reply:
I'll LGTM the API to unblock you.
Generally speaking, I suggest thinking about wording from the perspective of a user of Envoy the implementer of the side-stream server, as opposed to what's the current Envoy implementation.

Reply:
SG